### PR TITLE
Order a virtual or hardware server by frondend/backend using router or vlan ids.

### PR DESCRIPTION
**Pull request by FernandoOjeda**
Order a virtual or hardware server by frondend/backend using router or vlan ids.
#493 

Note: It seems that the SoftLayer go client does not support to order a vs using router ids, only by vlan ids.

### CODE REVIEW

File: content/go/order_server_by_vlan_router.go.md [hunk 1]
```diff
@@ -0,0 +1,790 @@
...
+
+    2. Use getServerPrices() to find the item keyNames you want to include in your order. These price IDs can be 
+    included prices array directly, but I’ve included gatherPriceIds() to match up KeyNames to build a list of price 
+    ids. getServerPrices() will also show the locations available for ordering.
+
+    3. listAvailableVlans() if you want to place the server on a specific VLAN.
+
+    4. listAvailableRouters() if you want to place the server on a specific router.
+
+```
```
[the remark this answers is not in the document]
Comment (ATGE) [reply content/go/order_server_by_vlan_router.go.md:34]: tagging code block as `go` helps to highlight the code properly.

**Review by allmightyspiff — DISMISSED** [review]
Looks good, address German's comments and this should be good to go.

Thanks.

File: content/go/order_server_by_vlan_router.go.md [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+	Remove the comment '//' below to order a bare metal server.
+	*/
+
+	locationKeyName := "DALLAS05"
+
+	OrderHardwareServer(packageId, locationKeyName, presetId, pubVlanId, privVlanId)
+
+}
+
+func OrderHardwareServer(packageId int, locationKeyName string, flavor int, frontend int, backend int) {
```
[the remark this answers is not in the document]
Comment (FernandoOjeda) [reply content/go/order_server_by_vlan_router.go.md]: Done.

File: content/go/order_server_by_vlan_router.go.md [hunk 3]
```diff
@@ -0,0 +1,790 @@
...
+
+	/**
+	Step 5, Order a virtual server by backend/frontend vlan.
+
+	Remove the comment '//' below to order a bare metal server.
+	*/
+
+	locationKeyName := "DALLAS05"
+
+	OrderHardwareServer(packageId, locationKeyName, presetId, pubVlanId, privVlanId)
```
[the remark this answers is not in the document]
Comment (FernandoOjeda) [reply content/go/order_server_by_vlan_router.go.md]: Done.

File: content/go/order_server_by_vlan_router.go.md [hunk 4]
```diff
@@ -0,0 +1,790 @@
...
+		items:     items,
+		location:  locationKeyName,
+		packageId: packageId,
+		flavor:    flavor,
+		hourly:    true,
+	}
+
+	prices := GetPrices(packageId, items)
+
+	// Build the Hardware Server array object.
```
[the remark this answers is not in the document]
Comment (FernandoOjeda) [reply content/go/order_server_by_vlan_router.go.md]: Done.

File: content/go/order_server_by_vlan_router.go.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (FernandoOjeda) [reply content/go/order_server_by_vlan_router.go.md]: Done.

File: content/go/order_server_by_vlan_router.go.md [hunk 5]
```diff
@@ -0,0 +1,790 @@
...
+
+    2. Use getServerPrices() to find the item keyNames you want to include in your order. These price IDs can be 
+    included prices array directly, but I’ve included gatherPriceIds() to match up KeyNames to build a list of price 
+    ids. getServerPrices() will also show the locations available for ordering.
+
+    3. listActivePresets() will list all active presets for a specific package.
+
+    4. listAvailableVlans() if you want to place the server on a specific VLAN.
+
+```
```
[the remark this answers is not in the document]
Comment (FernandoOjeda) [reply content/go/order_server_by_vlan_router.go.md]: Done.

**Review by ATGE — APPROVED** [review]

**Review by allmightyspiff — APPROVED** [review]
Looks good.

**Pull request merged**